src/net/rpc.rs

```diff
@@ -543,6 +548,131 @@ impl ConversationHttp {
+                });
+            match res {
+                Ok(inv_info) => {
+                    let current_page = inv_info.pages_count - 1;
+                    (current_page, current_page, vec![current_page])
+                }
+                Err(_) => {
+                    let msg = format!("Unable to read contract BNS");
+                    warn!("{}", msg);
+                    let response = HttpResponseType::ServerError(response_metadata, msg.clone());
+                    response.send(http, fd)?;
+                    return Err(net_error::ChainstateError(msg));
+                }
+            }
+        };
+
+        // We need to rebuild an ancestry tree, but we're still missing some informations at this point
+        let (min_block_height, max_block_height) = match atlasdb
+            .get_minmax_heights_window_for_page_index(oldest_page_index, newest_page_index)
+        {
+            Ok(window) => window,
+            Err(e) => {
+                let msg = format!("Unable to read Atlas DB");
+                warn!("{}", msg);
+                let response = HttpResponseType::ServerError(response_metadata, msg.clone());
+                response.send(http, fd)?;
+                return Err(net_error::DBError(e));
+            }
+        };
+
+        let mut blocks_ids = vec![];
+        let mut headers_tx = chainstate.headers_tx_begin()?;
+        let tip_index_hash =
+            StacksBlockHeader::make_index_block_hash(tip_consensus_hash, tip_block_hash);
+
+        for block_height in min_block_height..=max_block_height {
+            match StacksChainState::get_index_tip_ancestor(
+                &mut headers_tx,
+                &tip_index_hash,
+                block_height,
+            )? {
+                Some(header) => blocks_ids.push(header.index_block_hash()),
+                _ => {}
+            }
+        }
+
+        match atlasdb.get_attachments_available_at_pages_indexes(&pages_indexes, &blocks_ids) {
+            Ok(pages) => {
+                let pages = pages
+                    .into_iter()
+                    .zip(pages_indexes)
+                    .map(|(inventory, index)| AttachmentPage { index, inventory })
+                    .collect();
+
+                let content = GetAttachmentsInvResponse {
+                    block_id: tip.clone(),
+                    pages,
+                };
+                let response = HttpResponseType::GetAttachmentsInv(response_metadata, content);
+                response.send(http, fd)
+            }
+            Err(e) => {
+                let msg = format!("Unable to read Atlas DB");
+                warn!("{}", msg);
+                let response = HttpResponseType::ServerError(response_metadata, msg.clone());
+                response.send(http, fd)?;
+                return Err(net_error::DBError(e));
+            }
+        }
+    }
+
+    fn handle_getattachment<W: Write>(
```

Comment:
It doesn't have to be done in this PR, but can you update this method so that the code _streams_ the attachment back, instead of loading it all in one go?  This is what we do for blocks, for example.

Reply:
Sounds good, will address with https://github.com/blockstack/stacks-blockchain/issues/2077